src/Diagnostics/FxCop/System.Runtime.Analyzers/Core/Globalization/UseOrdinalStringComparison.Fixer.cs

```diff
@@ -0,0 +1,139 @@
+
+            // We cannot have multiple overlapping diagnostics of this id.
+            var diagnostic = context.Diagnostics.Single();
+
+            if (IsInArgumentContext(node))
+            {
+                // StringComparison.CurrentCulture => StringComparison.Ordinal
+                // StringComparison.CurrentCultureIgnoreCase => StringComparison.OrdinalIgnoreCase
+                context.RegisterCodeFix(new MyCodeAction(SystemRuntimeAnalyzersResources.StringComparisonShouldBeOrdinalOrOrdinalIgnoreCase,
+                                                         async ct => await FixArgument(context.Document, syntaxFactoryService, root, node).ConfigureAwait(false)),
+                                                    diagnostic);
+            }
+            else if (IsInIdentifierNameContext(node))
+            {
+                // string.Equals(a, b) => string.Equals(a, b, StringComparison.Ordinal)
+                // string.Compare(a, b) => string.Compare(a, b, StringComparison.Ordinal)
+                context.RegisterCodeFix(new MyCodeAction(SystemRuntimeAnalyzersResources.StringComparisonShouldBeOrdinalOrOrdinalIgnoreCase,
+                                                         async ct => await FixIdentifierName(context.Document, syntaxFactoryService, root, node, context.CancellationToken).ConfigureAwait(false)),
+                                                    diagnostic);
+            }
+            else if (IsInEqualsContext(node))
+            {
+                // "a == b" => "string.Equals(a, b, StringComparison.Ordinal)"
+                // "a != b" => "!string.Equals(a, b, StringComparison.Ordinal)"
+                context.RegisterCodeFix(new MyCodeAction(SystemRuntimeAnalyzersResources.StringComparisonShouldBeOrdinalOrOrdinalIgnoreCase,
+                                async ct => await FixEquals(context.Document, syntaxFactoryService, root, node, context.CancellationToken).ConfigureAwait(false)),
+                        diagnostic);
+            }
+        }
+
+        protected abstract bool IsInArgumentContext(SyntaxNode node);
+        protected abstract Task<Document> FixArgument(Document document, SyntaxGenerator syntaxFactoryService, SyntaxNode root, SyntaxNode argument);
+
+        protected abstract bool IsInIdentifierNameContext(SyntaxNode node);
+        protected abstract Task<Document> FixIdentifierName(Document document, SyntaxGenerator syntaxFactoryService, SyntaxNode root, SyntaxNode identifier, CancellationToken cancellationToken);
+
+        protected abstract bool IsInEqualsContext(SyntaxNode node);
+        protected abstract Task<Document> FixEquals(Document document, SyntaxGenerator syntaxFactoryService, SyntaxNode root, SyntaxNode node, CancellationToken cancellationToken);
+
+        internal SyntaxNode CreateEqualsExpression(SyntaxGenerator syntaxFactoryService, SemanticModel model, SyntaxNode operand1, SyntaxNode operand2, bool isEquals)
```

Comment:
Name for the method 'CreateEqualsExpression' is misleading. We are not creating an EqualsExpression here - we are replacing existing EqualsExpression with an invocation to String.Equals().